eth/downloader/queue.go

```diff
@@ -317,6 +351,45 @@ func (q *queue) Schedule61(hashes []common.Hash, fifo bool) []common.Hash {
 	return inserts
 }
 
+// ScheduleSkeleton adds a batch of header retrieval tasks to the queue to fill
+// up an already retrieved header skeleton.
+func (q *queue) ScheduleSkeleton(from uint64, skeleton []*types.Header) {
+	q.lock.Lock()
+	defer q.lock.Unlock()
+
+	// No skeleton retrieval can be in progress, fail hard if so (huge implementation bug)
+	if q.headerResults != nil {
+		panic("skeleton assembly already in progress")
```

Comment:
This check is mostly for code quality reasons, so that if we screw up the code in the future it fails hard. I don't think that given the stack traces would be enough to figure out the reason that lead to this scenario. We can change it of course, just seems a plain panic is cleaner that a log embedded one.